fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/CustomItemSetting.java

```diff
@@ -30,7 +30,26 @@
  * to items in a way that is compatible with other mods that add
  * settings to items.
  *
- * <p>Values of this setting can be retrieved from an item using {@link CustomItemSetting#getValue(Item)}</p>
+ * <p>Values of this setting can be retrieved from an item using {@link CustomItemSetting#getValue(Item)}
+ *
+ * <p>Users that wish to expose a custom setting for use in other mods should do so by exposing
+ * the CustomItemSetting instance.
+ *
+ * <pre>{@code
+ * public static final CustomItemSetting<String> CUSTOM_TOOLTIP = CustomItemSetting.create(() -> null);
+ *
+ * @Override
+ * public void onInitializeClient() {
+ *     ItemTooltipCallback.EVENT.register((stack, context, lines) -> {
+ *         String tooltip = CUSTOM_TOOLTIP.getValue(stack.getItem());
+ *
+ *         if (tooltip != null) {
+ *             lines.add(new LiteralText(tooltip));
+ *         }
+ *     }
+ * }}</pre>
```

Comment:
Thanks for adding this. Might want to explain a little bit what's going on here with a few comments, and show how it is used with the item settings, i.e. `// Create the setting instance`, `// Register a setting for an item` and finally `// Use the custom setting`. And possibly prefix with `<h3>Usage Example</h3>`.

Then we can also link to this from `FabricItemSettings.custom()`: `See {@link CustomItemSetting} for a full usage example.`

Yes, writing javadoc is pain, but it is the key to good API use by the community. Don't compromise on quality here because people are likely to copy/paste even if they shouldn't. 😄